### ISSUE TITLE
Add documentation page on migrating from R

### ISSUE THREAD

**Issue by Irio** [numpy/numpy#15792]
Prior of writing it, it should probably be defined the scope of this documentation page. In my understanding, based on the definition in [NEP 44](https://numpy.org/neps/nep-0044-restructuring-numpy-docs.html), that would be a Reference Guide.

I see two possible ways of putting it:

1. A short reference of commands and similarities/differences, similar to [NumPy for Matlab users](https://numpy.org/devdocs/user/numpy-for-matlab-users.html).
2. A longer reference that not only cites similarities/differences, but helps the user to be productive while following NumPy standards.

The befenits I see in the first option: it would be much shorter, thus possibly quicker to write and publish; also, with the right presentation, it should be easy to read, allowing users to quickly start using NumPy to accomplish their goals.

In favor of the second option, we have the assumption that a longer introduction to NumPy, SciPy, and Python would give them a more productive start. For instance, R users usually have [R Markdown](https://rmarkdown.rstudio.com/) right on their side, while NumPy users may decide to choose [Jupyter](https://jupyter.org/); dataframes are part of R, while NumPy users could do same things in pure NumPy or use [Pandas](https://pandas.pydata.org/) on top of it. With this in mind, the second option would contain an introduction to the SciPy ecossystem rather than be limited to NumPy.

Some ideas of topics:

* Philosophical differences between the two environments
* Data types
    * vectors, lists, and dataframes vs NumPy and Pandas equivalents
    * how to get the shape of an object
* Reading documentation of functions
    * `?`, `??`, and `help`
    * official websites
* Reproducible research
    * R Markdown vs Jupyter
* Distributions (RStudio vs Anaconda perhaps?)
* Indexing
    * subsetting
    * views
    * copy
* Element-wise operations
    * tell how NumPy will give them that, but pure Python might cause weird behavior
* Applying functions on elements/dimensions/array
* Descriptive statistics
* Other statistical functions/packages
* Reading/writing data to disk
* Plotting
* tydiverse vs SciPy
* Debugging/IDEs
* Interfacing between the two
* Deployment

Depending on the scope to be defined, some or few may be taken away from this main reference page. In that case, they might generate tutorials that would be linked for furter reading.

**Comment by rgommers** [comment]
This could be very useful. I'm slightly hesitant though, since I have the feeling it belongs more naturally in the Pandas than the NumPy docs.

> In my understanding, based on the definition in [NEP 44](https://numpy.org/neps/nep-0044-restructuring-numpy-docs.html), that would be a Reference Guide.

Reference Guide really is for the NumPy API docs. This would be a How-To I think ("how do I do X in Python vs. X in R").

**Comment by WarrenWeckesser** [comment]
> I'm slightly hesitant though, since I have the feeling it belongs more naturally in the Pandas than the NumPy docs.

Good point.  Pandas has a pretty extensive collection of docs, including [Comparison with R / R libraries](https://pandas.pydata.org/docs/getting_started/comparison/comparison_with_r.html).

**Comment by melissawm** [comment]
I agree that the content seems to make more sense for Pandas than NumPy. Perhaps you can submit this content there, @Irio , as a How-To or Tutorial. We can add a link somewhere in our docs making it clear that if people are coming from R they probably will need Pandas, and send them to your document there. What do you think?

**Comment by rossbar** [comment]
The consensus seems to be that the pandas docs are a better fit for this article (maybe just expanding on already-existing documentation there). This is a really nice summary at the top though, so I'm hesitant to close this :)

**Comment by rgommers** [comment]
@noatamir what do you think about migrating this to the Pandas issue tracker? Would there be interest in Pandas to expand on its current R comparison docs?
